pyro/infer/mcmc/trace_kernel.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import absolute_import, division, print_function
+
+from abc import abstractmethod, ABCMeta
+
+from six import add_metaclass
+
+
+@add_metaclass(ABCMeta)
+class TraceKernel(object):
+
+    def setup(self, *args, **kwargs):
+        """
+        Optional method to set up any state required before the run.
+
+        :param \*args: Algorithm specific positional arguments.
+        :param \*\*kwargs: Algorithm specific keyword arguments.
+        """
+        pass
+
+    def cleanup(self):
+        """
+        Optional method to clean up any residual state on termination.
+        """
+        pass
+
+    def diagnostics(self, time_step):
+        """
```

Comment:
Removed `num_accepts` from #579 in favor of a general diagnostics method so that the implementing classes are free to provide whatever diagnostic information may be useful.